src/site/_Site.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import {
+  BrowserRouter, Switch, Route, Redirect,
+} from 'react-router-dom';
+import PropTypes from 'prop-types';
+import ModalManager from 'terra-application/lib/modal-manager';
+
+import DevSiteNavigation from '../navigation/_DevSiteNavigation';
+import Raw from '../raw/_Raw';
+import AppSettingsContext from '../navigation/_AppSettingsContext';
+import siteConfigPropType from './siteConfigPropTypes';
+
+import './site.module.scss';
+
+const propTypes = {
+  /**
+   * Render prop for rendering terra-application-base or a variant of it.
+   */
+  applicationBase: PropTypes.func.isRequired,
+
+  /**
+   * Render prop for rendering application navigation or a variant of it.
+   */
+  applicationNavigation: PropTypes.func.isRequired,
+
+  /**
+   * basename is expected to be '' or '/*', used for react router
+   */
+  basename: PropTypes.string.isRequired,
+
+  /**
+   * The site config for the application.
+   */
+  siteConfig: siteConfigPropType.isRequired,
+
+  /**
+   * function to return search items
+   */
+  fetchSearchItems: PropTypes.func,
```

Comment:
Technically not required if you wanted to remove the site search extension.

Reply:
I think this needs to be updated to prevent prop type errors then: https://github.com/cerner/terra-dev-site/pull/175/files?file-filters%5B%5D=.html&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.jsx&file-filters%5B%5D=.md&file-filters%5B%5D=.png&file-filters%5B%5D=.scss&file-filters%5B%5D=.yml&file-filters%5B%5D=dotfile&hide-deleted-files=true&short_path=8d6bac4#diff-db0b9ed73435818d9fd24581c66f8c9cR34

Reply:
fair enough, for now it's required here.